### PR TITLE
[Data Objects] Fixed inherited classification-store data is not displ…

### LINKED ISSUE #6029: [Data Objects] ClassificationStore grid view inherited data
For a future customer project, the classification store in the grid view must work with inheritance. Unfortunately this is not yet happening.

I already fixed that in the forked-repo and I will push that right now.

### PR DESCRIPTION
**Pull request by skochTC**
fixes #6029

### CODE REVIEW

**Comment by weisswurstkanone** [comment]
Hi @skochTC , can you provide steps to reproduce + point out the significant change as some restructuring happened as well.
Thanks!

**Comment by skochTC** [comment]
> 
> 
> Hi @skochTC , can you provide steps to reproduce + point out the significant change as some restructuring happened as well.
> Thanks!

Hi,

to report this you need to create a store with some keys. Make sure that inheritance is enabled in the class definition. Now create a parent and a child element. The inherited data of the parent element is displayed in the object view in the child. But they are not displayed in the GRID view.

For a better explanation, see the following screenshot:
image

After my changes, the inherited data from the store is displayed correctly in the GRID view:
image

> point out the significant change as some restructuring happened as well.

I'm not really sure I got that right. I' ve added a comment in the code

File: models/DataObject/Service.php [hunk 1]
```diff
@@ -446,6 +417,16 @@ public static function gridObjectData($object, $fields = null, $requestedLanguag
                         }
                     }
 
+                    // because the key for the classification store has not a direct getter, you have to check separately if the data is inheritable
+                    if (strpos($key, '~') === 0 && empty($data[$key])) {
```
Comment (weisswurstkanone) [inline models/DataObject/Service.php]: guess the data-types isEmpty method should be used instead.

does (strpos($key, '~') === 0 also work for bricks and so on ?

File: models/DataObject/Service.php [hunk 2]
```diff
@@ -724,6 +705,62 @@ private static function getValueForObject($object, $key, $brickType = null, $bri
...
+                    if (method_exists($definition, 'getDataForGrid')) {
+                        $fielddata = $definition->getDataForGrid($fielddata, $object);
+                    }
+
+                    return $fielddata;
+                }
+            }
+        }
+
+        return '';
```
Comment (weisswurstkanone) [inline models/DataObject/Service.php]: returning an empty string here is changing the original behavior

File: models/DataObject/Service.php (hunk 1 above)
Comment (skochTC) [reply models/DataObject/Service.php]: > guess the data-types isEmpty method should be used instead.

For Classification-Store keys `$def` is `null`. So `isEmpty` does not work.

> does (strpos($key, '~') === 0 also work for bricks and so on ?

Right, I'll fix that right now.

File: models/DataObject/Service.php (hunk 2 above)
Comment (skochTC) [reply models/DataObject/Service.php]: Okay in order to not change the original behavior, I return "null" like it was before.

File: models/DataObject/Service.php (hunk 1 above)
Comment (skochTC) [reply models/DataObject/Service.php]: Fixed in commit: 3e03840b05b74a5aa98a9440d891d492bb4307ad

**Comment by weisswurstkanone** [comment]
thanks!

**Pull request merged**